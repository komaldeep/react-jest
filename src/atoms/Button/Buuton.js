import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './Buuton.css';

export const buttonSizes = {
  LG: 'lg',
  MD: 'md',
  SM: 'sm',
  XS: 'xs',
};

export const buttonSkins = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
};

export default class Buuton extends Component {

  static displayName = 'atoms/Button/Buuton.js';

  static propTypes = {
    size: PropTypes.string,
    skin: PropTypes.string,
    disabled: PropTypes.bool,
    onClick: PropTypes.func,
    children: PropTypes.node,
    block: PropTypes.bool,
    type: PropTypes.oneOf(['button', 'submit']),
  };

  static defaultProps = {
    skin: buttonSkins.PRIMARY,
    size: buttonSizes.MD,
    onClick: () => {},
    disabled: false,
    block: false,
    type: 'button'
  };

  constructor (props) {
    super(props)
    this.state = {
      counter:0
    }
  }

  handleClick = () => {
    this.setState({
      counter:1
    })
  };

  render() {
    return (
      <button
        onClick={ this.handleClick }
        className={
          classNames({
            btn: true,
            large: this.props.size === 'lg',
            small: this.props.size === 'sm'
          })
        }
        disabled={ this.props.disabled }
        type={this.props.type}
      >
        { this.props.children }
      </button>
    );
  }
}
