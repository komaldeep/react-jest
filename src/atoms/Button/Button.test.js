import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json';
import Buuton, { buttonSkins, buttonSizes } from './Buuton';

test('renders without crashing', () => {
  const rendered = renderer.create(<Buuton />).toJSON();
  expect(rendered).toMatchSnapshot();
});

test('giving specifioc props to button', () => {
  const component = shallow(<Buuton
    size={buttonSizes.LG}
    skin={buttonSkins.PRIMARY}
    disabled={ false }
    block = {false}
    type='button'>
    This is button
    </Buuton>)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot();
});

test('on click of button', () => {
  const component = mount(<Buuton
    size={buttonSizes.SM}
    skin={buttonSkins.SECONDARY}>
     button with different features
  </Buuton>)
  const button = component.find('button');
  button.simulate('click');
  expect(component.state().counter).toEqual(1);
})

it('should render a right size and skin', () => {
  const wrapper = mount(
    <Buuton
      size={buttonSizes.SM}
      skin={buttonSkins.SECONDARY}>
      button with different features
    </Buuton>
  );
  expect(wrapper.prop('size')).toEqual('sm');
  expect(wrapper.prop('skin')).toEqual('secondary');
})
