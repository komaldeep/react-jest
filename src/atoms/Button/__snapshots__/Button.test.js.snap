// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<button
  className="btn"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;
